File: frontend/src/types/questao.ts
export type TipoQuestao = 'multipla-escolha' | 'ordenacao' | 'preencher-lacuna'

export interface Questao {
  id: string
  tipo: TipoQuestao
  enunciado: string
  codigo?: string
  opcoes?: string[]
  respostaCorreta: string | string[]
}

File: backend/app/schemas/questao.py
from pydantic import BaseModel

from app.models.enums import TipoAtividade

__all__ = ["Questao", "TipoAtividade"]


class Questao(BaseModel):
    id: str
    tipo: TipoAtividade
    enunciado: str
    codigo: str | None = None
    opcoes: list[str] | None = None
    resposta_correta: str | list[str]
